### PR TITLE
[MRG] FIX section ordering if pre-defined section was used

### PR DESCRIPTION
**Pull request by mainakjas**
Closes https://github.com/mne-tools/mne-python/issues/1882

### CODE REVIEW

**Comment by dengemann** [comment]
I'm -2 against this solution. I'll try to meet with @mainakjas this week to think about a clean way separate parsing from scripting mode.

**Comment by mainakjas** [comment]
Are we good now @dengemann ? Let me know.

File: mne/report.py [hunk 1]
```diff
@@ -944,7 +947,7 @@ def _init_render(self, verbose=None):
 
     @verbose
     def parse_folder(self, data_path, pattern='*.fif', n_jobs=1, mri_decim=2,
-                     verbose=None):
+                     sort=True, verbose=None):
```
Comment (agramfort) [inline mne/report.py]: I would call it sort_sections

**Comment by mainakjas** [comment]
done @agramfort

File: mne/report.py [hunk 2]
```diff
@@ -671,6 +671,9 @@ def __init__(self, info_fname=None, subjects_dir=None, subject=None,
         self.sections = []  # List of sections
         self._sectionlabels = []  # Section labels
         self._sectionvars = {}  # Section variable names in js
+        # boolean to specify if sections should be ordered in natural
+        # order of processing (raw -> events ... -> inverse)
+        self._sort_sections = False
 
         self._init_render()  # Initialize the renderer
```
Comment (dengemann) [inline mne/report.py]: @mainakjas this will be False by default when the report is constructed?

Comment (mainakjas) [inline mne/report.py]: yes. And it's made `True` only if you call `parse_folder`

**Comment by Eric89GXL** [comment]
Is this still DONT MERGE, or is it good now?

**Comment by dengemann** [comment]
With @mainakjas and @agramfort we found a different solution to this. Closing.

**Comment by mainakjas** [comment]
@dengemann I pushed the solution that we discussed to this PR.

**Comment by dengemann** [comment]
@mainakjas ouupsy.

**Comment by dengemann** [comment]
+1 for merge once Travis is happy

**Comment by Eric89GXL** [comment]
Coverage decrease boo hiss

I'm actually not sure how that could happen from this PR...

**Comment by Eric89GXL** [comment]
Thanks @mainakjas!

**Comment by dengemann** [comment]
Thanks @mainakjas !

**Pull request merged**